onnx/defs/nn/defs.cc

```diff
@@ -2228,6 +2228,61 @@ ONNX_OPERATOR_SET_SCHEMA(
         })
         .SetDoc(TfIdfVectorizer_ver9_doc));
 
+static const char* StringSplit_doc =
+    R"DOC(StringSplit splits a string tensor based on a delimiter attribute and a maxsplit attribute. The output of this operator is a (potentially nested) Sequence of tensors of strings. The shape of the output nested Sequence is the same as the input tensor shape, and the string tensors contain the substrings split by the delimiter at the same position as in the input.)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    StringSplit,
+    20,
+    OpSchema()
+        .Input(0, "X", "String Tensor to split", "T", OpSchema::Single, true, 1, OpSchema::NonDifferentiable)
+        .Attr(
+            "delimiter",
+            "Delimiter to split on. If left unset this defaults to a space character.",
+            AttributeProto::STRING,
+            false)
+        .Attr(
+            "maxsplit",
+            "Maximum number of splits. If left unset, it will make as many splits as many times the delimiter appears.",
+            AttributeProto::INT,
+            false)
+        .Output(
+            0,
+            "Y",
+            "Sequence of split strings (with equal Sequence shape as the input tensor)",
+            "S",
+            OpSchema::Single,
+            true,
+            1,
+            OpSchema::NonDifferentiable)
+        .TypeConstraint("T", {"tensor(string)"}, "The input must be a UTF-8 string tensor")
+        .TypeConstraint(
+            "S",
+            {"seq(tensor(string))", "seq(seq(tensor(string)))"},
```

Comment:
Currently, this string representation of the types is used inside the [shape inference logic](https://github.com/onnx/onnx/blob/2b7ce71ca46fedba829e592ae5c5e6415fdde105/onnx/defs/schema.cc#L94-L100) to make sure we don't have an unexpected type - this constraint in the shape inference code means that we cannot return sequences with an arbitrary amount of nesting, which I'd be open to changing.